diff -uprN linux-4.19.1/arch/arm/boot/dts/am335x-wega-bw.dts linux-4.19.1-patch/arch/arm/boot/dts/am335x-wega-bw.dts
--- linux-4.19.1/arch/arm/boot/dts/am335x-wega-bw.dts	1970-01-01 01:00:00.000000000 +0100
+++ linux-4.19.1-patch/arch/arm/boot/dts/am335x-wega-bw.dts	2017-06-20 11:41:15.833486302 +0200
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2018 IT-Klinger
+ * Author: Andreas Klinger <ak@it-klinger.de>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/dts-v1/;
+
+#include "am335x-phycore-som.dtsi"
+#include "am335x-wega.dtsi"
+
+/ {
+	regulators {
+		compatible = "simple-bus";
+
+		avdd: avdd {
+			compatible = "regulator-fixed";
+			regulator-name = "avdd";
+			regulator-min-microvolt = <2600000>;
+			regulator-max-microvolt = <2600000>;
+			regulator-boot-on;
+		};
+	};
+};
+
+&am33xx_pinmux {
+	weight0_pins: weight0_pins {
+		pinctrl-single,pins = <
+			AM33XX_IOPAD(0x828, PIN_OUTPUT | MUX_MODE7) /* SCK */
+			AM33XX_IOPAD(0x834, PIN_INPUT | MUX_MODE7)  /* DOUT */
+			>;
+	};
+};
+
+/* SoM */
+&i2c_eeprom {
+	status = "okay";
+};
+
+&i2c_rtc {
+	status = "okay";
+};
+
+/ {
+	ocp {
+		weight0 {
+			pinctrl-names = "default";
+			pinctrl-0 = <&weight0_pins>;
+			compatible = "avia,hx711";
+			sck-gpios = <&gpio0 26 GPIO_ACTIVE_HIGH>;
+			dout-gpios = <&gpio1 13 GPIO_ACTIVE_HIGH>;
+			avdd-supply = <&avdd>;
+			clock-frequency = <100000>;
+		};
+	};
+};
+
diff -uprN linux-4.19.1/arch/arm/boot/dts/Makefile linux-4.19.1-patch/arch/arm/boot/dts/Makefile
--- linux-4.19.1/arch/arm/boot/dts/Makefile	2017-06-17 06:47:27.000000000 +0200
+++ linux-4.19.1-patch/arch/arm/boot/dts/Makefile	2017-06-20 11:42:00.909488918 +0200
@@ -716,5 +716,6 @@ dtb-$(CONFIG_SOC_AM33XX) += \
	am335x-shc.dtb \
	am335x-sbc-t335.dtb \
	am335x-sl50.dtb \
+	am335x-wega-bw.dtb \
	am335x-wega-rdk.dtb \
	am335x-osd3358-sm-red.dtb
